### PR TITLE
Fixes Telescope #2134: Move Elastic client to Satellite

### PR DESCRIPTION
**Pull request by HyperTHD**
This PR refactors the elastic client code from Telescope into Satellite. This fulfills PR [#2134](https://github.com/Seneca-CDOT/telescope/issues/2134).

Currently working on the tests, should soon be ready for review

### CODE REVIEW

File: src/elastic.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+const Mock = require('@elastic/elasticsearch-mock');
+
+function MockClient(options) {
+  const mock = new Mock();
+  options.Connection = mock.getConnection();
+  // Mock out various responses we'll need:
+  mock.add(
+    {
+      method: ['PUT', 'POST', 'GET', 'DELETE'],
+      path: '/posts/post/:post_id',
```
Comment (humphd) [inline src/elastic.js:12]: `path: '*',`

File: src/elastic.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  return client;
+}
+
+// Set MOCK_ELASTIC=1 to mock, MOCK_ELASTIC= to use real elastic
+const useMockElastic = process.env.MOCK_ELASTIC;
+
+// Use either a real Elastic Client or a Mock instance, depending on env setting
+const ElasticConstructor = useMockElastic ? MockClient : Client;
+
+function createElasticClient() {
```
Comment (humphd) [inline src/elastic.js:32]: Let's allow passing in options here, and pass them through:

```js
function createElasticClient(options = {}) {
  const elasticUrl = `${ELASTIC_URL}:${ELASTIC_PORT}`;
  return new ElasticConstructor({ ...options, node: elasticUrl });
}
```

**Review by humphd — CHANGES_REQUESTED** [review]

File: src/elastic.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    {
+      method: ['PUT', 'POST', 'GET', 'DELETE'],
+      path: '*',
+    },
+    () => {
+      return { status: 'ok' };
+    }
+  );
+
+  const client = new Client(options);
```
Comment (humphd) [inline src/elastic.js:18]: I think you need to do what the mock client does with Elastic to set the mock connection on options:

```js
const { Client } = require('@elastic/elasticsearch')
const Mock = require('@elastic/elasticsearch-mock')
 
const mock = new Mock()
const client = new Client({
  node: 'http://localhost:9200',
  Connection: mock.getConnection()
})
```

File: test.js [hunk 4]
```diff
@@ -903,3 +904,15 @@ describe('Redis()', () => {
     });
   });
 });
+
+describe('Elastic()', () => {
+  test('Testing the name property which should be a string', (done) => {
+    const elastic = Elastic({ name: 'client-test' });
+
+    elastic.info(console.log);
```
Comment (humphd) [inline test.js:912]: Assert the values that `elastic.info()` produces vs passing to console.log

File: test.js [hunk 5]
```diff
@@ -903,3 +904,15 @@ describe('Redis()', () => {
...
+
+    elastic.end();
+
+    done();
```
Comment (humphd) [inline test.js:916]: Unless you're inside a callback (which you might want to be with my change above), you don't need `done()` in sync paths.

**Review by humphd — CHANGES_REQUESTED** [review]

**Review by humphd — CHANGES_REQUESTED** [review]
Excellent.  Please update the README with details about this, matching what we do there now for Redis.

**Review by humphd — APPROVED** [review]

**Review by Metropass — APPROVED** [review]

**Comment by humphd** [comment]
I've tagged v1.13.0 to include this, which is building now.

**Pull request merged**